ISISNeutronMuon/MDANSE
# Remove ability to plot non-meaningful data

**Description of work.**

Fixes #24 

Currently, it is possible to plot all variables in a NetCDF file, including those that have no use or even those that do not make sense, such as the description of the trajectory, or time against atomic mass. Several such variables have been prevented from being displayed in the list of variables. If anyone knows of other variables which should not be available for plotting, please let me know.

Furthermore, when data containing non-numeric values is attempted to be plotted, it now results in an error with a meaningful message without actually plotting the data.

**To test:**

1. Load an .nc file containing a trajectory into 2D/3D Plotter.
2. Observe that there are no variables titled 'description', 'step', or 'time' available for plotting.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**.
